vm/systemSmartContracts/validator.go

```diff
@@ -1124,9 +1127,35 @@ func (v *validatorSC) unStakeNodesFromStakingSC(blsKeys [][]byte, registrationDa
 		}
 
 		numSuccess++
+
+		stakedData, err := v.getStakedData(blsKey)
+		if err != nil {
+			continue
+		}
+		if stakedData.UnStakedNonce > currentNonce {
```

Comment:
As found out by @AdoAdoAdo & me this is always 0 if the bls key was never staked. UnStakedEpoch will hold core.DefaultUnstakedEpoch

Reply:
fixed this.